test/configuration_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ public:
+  MetadataAgentConfiguration config;
+  static void ParseConfiguration(MetadataAgentConfiguration& config, const std::string configString) {
+    std::stringstream stream(configString);
+    config.ParseConfiguration(stream);
+  }
+};
+
+static void TestDefaultConfig(MetadataAgentConfiguration& config) {
```

Comment:
Same here, you could move it back into the class as an instance method and have it access the `config` instance variable instead.